modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,703 @@
+##' call rp_control (from RpTools py package) from PEcAn workflow and store the output in BETY
+##'
+##' @name remote_process
+##' @title remote_process
+##' @export
+##' @param settings PEcAn settings list containing remotedata tags: raw_mimetype, raw_formatname, geofile, source, collection, scale, projection, qc, algorithm, credfile, pro_mimetype, pro_formatname, out_get_data, out_process_data
+##' @examples
+##' \dontrun{
+##' remote_process(settings)
+##' }
+##' @author Ayush Prasad
+##'
+
+remote_process <- function(settings) {
+  # information about the date variables used in remote_process -
+  # req_start, req_end : start, end dates requested by the user, the user does not have to be aware about the status of the requested file in the DB
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to remote_process for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+  
+  RpTools <- reticulate::import("RpTools")
+  
+  input_file <- NULL
+  stage_get_data <- NULL
+  stage_process_data <- NULL
+  raw_merge <- NULL
+  pro_merge <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
+  
+  # extract the variables from the settings list
+  siteid <- as.numeric(settings$run$site$id)
+  siteid_short <- paste0(siteid %/% 1e+09, "-", siteid %% 1e+09)
+  raw_mimetype <- settings$remotedata$raw_mimetype
+  raw_formatname <- settings$remotedata$raw_formatname
```

Comment:
I think we should switch to using registration files like Mike suggested. Many of these arguments could be extracted from those files, we might as well replace the lookup table for function names. Could be one registration file per source, e.g. for register.GEE.xml:
```
<?xml version="1.0"?>
<GEE>
  <collection>
    <original_name>COPERNICUS/S2_SR</original_name>
    <pecan_name>s2</pecan_name>
    <scale>site</scale>
    <format>
     <id></id>
     <name></name>
     <mimetype>application/x-netcdf</mimetype>
   </format>
  </collection>
  <collection>
    <original_name>LANDSAT/LC08/C01/T1_SR</original_name>
    <pecan_name>l8</pecan_name>
    <scale>site</scale>
    <format>
     <id></id>
     <name></name>
     <mimetype>application/x-netcdf</mimetype>
   </format>
  </collection>
    <collection>
    <original_name>NASA_USDA/HSL/SMAP_soil_moisture</original_name>
    <pecan_name>smap</pecan_name>
    <scale>site</scale>
    <format>
     <id></id>
     <name></name>
     <mimetype>application/x-netcdf</mimetype>
   </format>
  </collection>
</GEE>
```
I will work through an example as I go